### DISCUSSION TITLE
EventToCommandBehavior needs to cast to system.EventArgs

### DISCUSSION THREAD

**Discussion by mdelgadov** [CommunityToolkit/Maui.Markup#308] [Q&A] [answered]
In net 8 Markup 4.1 (Didn't try Net 9, too buggy yet)

In XAML, an eventtocommand behavior needs a cast:

 <Entry
                            x:Name="Email"
                            Grid.ColumnSpan="2"
                            Placeholder="{x:Static languages:Resources.Email}"
                            Text="{Binding Email, Mode=TwoWay}">
                            <Entry.Keyboard>
                                <Keyboard x:FactoryMethod="Create">
                                    <x:Arguments>
                                        <KeyboardFlags>Suggestions</KeyboardFlags>
                                    </x:Arguments>
                                </Keyboard>
                            </Entry.Keyboard>
                            <Entry.Behaviors>
                                <mct:EventToCommandBehavior
                                    **_x:TypeArguments="system:EventArgs"_**
                                    Command="{Binding UnfocusedEmailCommand}"
                                    EventName="Unfocused" />
                            </Entry.Behaviors>
                        </Entry>

In Markup, all I can find is something like this:

var behavior = new EventToCommandBehavior
        {
            EventName = nameof(CollectionView.SelectionChanged),
            Command = _vm.TappedCommand,
            CommandParameter = new Binding("SelectedItem", source: collectionView)
        };

Where the binding will take an element selected in a collection view.
It will compile but will have the error:

Parameter "parameter" (object) cannot be of type Microsoft.Maui.Controls.Binding, as the command type requires an argument of type System.EventArgs. (Parameter 'parameter')

That happens very deep in the stack in, at least, Windows and Android (haven't tested in iOS and MacCat)

This seems quite similar to the x:TypeArguments noted above.

Questions:

How can I cast in Markup the same way I'm doing it in Xaml?

Or

How can I pass a command parameter that works?

[DashboardMarkup.zip](https://github.com/user-attachments/files/16826132/DashboardMarkup.zip)
[DashboardVM.zip](https://github.com/user-attachments/files/16826136/DashboardVM.zip)
[monthly_sales.zip](https://github.com/user-attachments/files/16826548/monthly_sales.zip)

<img width="1290" alt="stack" src="https://github.com/user-attachments/assets/10d6d4ff-cf5c-4877-9f67-4d9142877b54">

**Comment by bijington** [comment]
You shouldn't need to set the CommandParameter property as per this example in our docs: https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/behaviors/event-to-command-behavior#c-markup

**Reply by mdelgadov** [reply, marked as answer]
Ok, I found the converters and they work for tapped and selected.
How do they work for Unfocused and TextChanged?

Do I need to extend them with a new converter?

**Reply by bijington** [reply]
Can't you achieve something like:

```csharp
public MainPage()
{
      InitializeComponent();

      var entry = new Entry();

      TextChangedCommand = new Command<TextChangedEventArgs>(args =>
      {

      });
      
      var behavior = new EventToCommandBehavior<TextChangedEventArgs>
      {
        EventName = nameof(Entry.TextChanged),
        Command = TextChangedCommand
      };
}
    
public ICommand TextChangedCommand { get; }
```
